test/e2e/tests/confirmations/signatures/signature-request.spec.ts

```diff
@@ -0,0 +1,269 @@
+          await verifySignatureResult(driver, data.type, publicAddress);
+        },
+      );
+    });
+  });
+
+  testData.forEach((data) => {
+    it(`can initiate and reject a Signature Request of ${data.type}`, async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .build(),
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver }) => {
+          await unlockWallet(driver);
+
+          await openDapp(driver);
+
+          // creates a sign typed data signature request
+          await triggerSignatureRequest(driver, data.type);
+
+          await driver.waitUntilXWindowHandles(3);
+          let windowHandles = await driver.getAllWindowHandles();
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.Dialog,
+            windowHandles,
+          );
+
+          // Reject signing typed data
+          await finalizeSignatureRequest(driver, 'Reject');
+          await driver.waitUntilXWindowHandles(2);
+          windowHandles = await driver.getAllWindowHandles();
+
+          // switch to the Dapp and verify the rejection was successful
+          await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+          await verifySignatureRejection(driver, data.type);
+        },
+      );
+    });
+  });
+
+  testData.forEach((data) => {
+    it(`can queue multiple Signature Requests of ${data.type} and reject`, async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .build(),
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver }) => {
+          const confirmation = new SignTypedData(driver);
+          await unlockWallet(driver);
+
+          await openDapp(driver);
+
+          // creates multiple sign typed data signature requests
+          await triggerSignatureRequest(driver, data.type);
+
+          await driver.waitUntilXWindowHandles(3);
+          const windowHandles = await driver.getAllWindowHandles();
+          // switches to Dapp
+          await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+          // creates second sign typed data signature request
+          await triggerSignatureRequest(driver, data.type);
+
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.Dialog,
+            windowHandles,
+          );
+
+          await confirmation.check_pageNumbers(1, 2);
+
+          // reject first signature request
+          await finalizeSignatureRequest(driver, 'Reject');
+          await driver.waitUntilXWindowHandles(3);
+
+          // reject second signature request
+          await finalizeSignatureRequest(driver, 'Reject');
+          await driver.waitUntilXWindowHandles(2);
```

Comment:
same as above: instead of mitigating race conditions at the spec level, could we maybe add some robustness in the finalizeSignatureRequest , method, by waiting for the window to close, when we expect so ie clickElementAndWaitForWindowToClose ?
in the first case, we don't wait for the dialog to close, but in this second one yes - so we could pass a param in the function